File: 2024_java_basics/14-nested-loops-lab/tasks/03-sum-prime-non-prime.java
import java.util.Scanner;

public class SumPrimeNonPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int sumPrime = 0;
        int sumNotPrime = 0;
        
        while (true) {
            String input = scanner.nextLine();
            
            if (input.equals("stop")) {
                break;
            }

            int n = Integer.parseInt(input);

            if (n < 0) {
                System.out.println("Number is negative.");
                continue;
            }

            if (n == 0) {
                continue;
            }

            boolean isPrime = true;
            
            if (n == 1) {
                isPrime = false;
            } else {
                for (int i = 2; i <= Math.sqrt(n); i++) {
                    if (n % i == 0) {
                        isPrime = false;
                        break;
                    }
                }
            }

            if (isPrime) {
                sumPrime += n;
            } else {
                sumNotPrime += n;
            }
        }

        System.out.printf("Sum of all prime numbers is: %d%n", sumPrime);
        System.out.printf("Sum of all non prime numbers is: %d%n", sumNotPrime);
        
        scanner.close();
    }
}

File: 2022_c#_basics/14-nested-loops-exercise/03-sum-prime-non-prime.cs
using System;

namespace _03.SumPrimeNonPrime
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();

            int sumPrime = 0;
            int sumNotPrime = 0;
                       
            while (input != "stop")
            {
                int n = int.Parse(input);
                int prime = 0;

                if (n < 0)
                {
                    Console.WriteLine("Number is negative.");

                    input = Console.ReadLine();
                    continue;
                }

                if (n == 0)
                {
                    input = Console.ReadLine();
                    continue;
                }

                for (int i = 1; i <= n; i++)
                {
                    if (n % i == 0)
                    {
                        prime++;
                    }
                }

                if (prime == 2)
                {
                    sumPrime += n;
                }
                else
                {
                    sumNotPrime += n;
                }

                input = Console.ReadLine();
            }

            Console.WriteLine($"Sum of all prime numbers is: {sumPrime}");
            Console.WriteLine($"Sum of all non prime numbers is: {sumNotPrime}");
        }
    }
}
